src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.cs

```diff
@@ -130,6 +131,13 @@ protected override void InitializeWorker(AnalysisContext context)
                 return null;
             }
 
+            // If the target method is writeable then we need to make sure we do not change a writeable 
+            // indexer to a read-only indexer, otherwise it will cause compiler errors
+            if (invocation.TargetMethod.IsReadOnly == false && invocation.Parent != null && invocation.Syntax.IsLeftSideOfAnyAssignExpression())
+            {
+                return null;
+            }
```

Comment:
i don't think this is the right fix.  The check we should be making is that the original Slice/indexer match in by-ref semantics.